CustomEd.LearningEngine.Service/Controllers/LearningEngineController.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            _studentRepository = studentRepository;
+            _chatbotMessageRepository = chatbotMessageRepository;
+            _learningPathRepository = learningPathRepository;
+            _httpContextAccessor = httpContextAccessor;
+            _jwtService = jwtService;
+            _mapper = mapper;
+        }
+
+        [Authorize(Policy = "StudentOnlyPolicy")]
+        [HttpGet("learningPaths")]
+        public async Task<ActionResult<SharedResponse<IEnumerable<LearningPath>>>> GetAllMyLearningPaths()
+        {
+            var studentId = new IdentityProvider(_httpContextAccessor, _jwtService).GetUserId();
+            if(studentId == Guid.Empty)
+            {
+                return Unauthorized(SharedResponse<IEnumerable<LearningPath>>.Fail("You're not authorized", null));
+            }
+            var learningPaths = await _learningPathRepository.GetAllAsync(x => x.StudentId == studentId);
+            return Ok(SharedResponse<IEnumerable<LearningPath>>.Success(learningPaths, null));
+        }
+
+         
+        [Authorize(Policy = "StudentOnlyPolicy")]
+        [HttpGet("learningPaths/{id}")]
+        public async Task<ActionResult<SharedResponse<LearningPath>>> GetLearningPath(Guid id)
+        {
+            var learningPath = await _learningPathRepository.GetAsync(id);
+
+            if (learningPath == null)
+            {
+                return NotFound(SharedResponse<LearningPath>.Fail("No learning path with such id", null));
+            }
+            return Ok(SharedResponse<LearningPath>.Success(learningPath, null));
+        }
+
+        [Authorize(Policy = "StudentOnlyPolicy")]
+        [HttpPost("learningPaths")]
+        public async Task<ActionResult<SharedResponse<LearningPath>>> CreateLearningPath(LearningPath learningPath)
+        {
+            var studentId = new IdentityProvider(_httpContextAccessor, _jwtService).GetUserId();
+            if(studentId == Guid.Empty)
+            {
+                return Unauthorized(SharedResponse<IEnumerable<LearningPath>>.Fail("You're not authorized", null));
+            }
+            learningPath.StudentId = studentId;
+            learningPath.Id = Guid.NewGuid();
+            await _learningPathRepository.CreateAsync(learningPath);
+            return Ok(SharedResponse<LearningPath>.Success(learningPath, null));
+        }
+
+        [Authorize(Policy = "StudentOnlyPolicy")]
+        [HttpPut("learningPaths")]
+        public async Task<ActionResult<SharedResponse<LearningPath>>> UpdateLearningPathStatus(Guid learningPathId, LearningPathStatus status)
+        {
+            var learningPath = await _learningPathRepository.GetAsync(learningPathId);
+            learningPath.Status = status;
+            await _learningPathRepository.UpdateAsync(learningPath);
+            return Ok(SharedResponse<LearningPath>.Success(learningPath, null));
+        }
+
+
+        [Authorize(Policy = "StudentOnlyPolicy")]
+        [HttpDelete("learningPaths/{id}")]
```

Comment:
routes should not be in camel case since uris are case insensitve. user learning-path instead